code/Ui/Win32/PathDialogWin32.cpp

```diff
@@ -14,42 +14,68 @@ namespace traktor::ui
 {
 
 PathDialogWin32::PathDialogWin32(EventSubject* owner)
-:	m_owner(owner)
+	: m_owner{ owner }
+	, m_hWnd{ nullptr }
+	, m_openDialogOptions{ 0 }
+	, m_pFileDialog{ nullptr }
+	, m_title{ 0 }
 {
 }
 
 bool PathDialogWin32::create(IWidget* parent, const std::wstring& title)
 {
+	m_hWnd = (HWND)parent->getInternalHandle();
 	_tcscpy_s(m_title, sizeof_array(m_title), wstots(title).c_str());
 
-	std::memset(&m_bi, 0, sizeof(m_bi));
-	m_bi.hwndOwner = parent ? (HWND)parent->getInternalHandle() : NULL;
-	m_bi.pidlRoot = NULL;
-	m_bi.pszDisplayName = m_title;
-	m_bi.lpszTitle = m_title;
-	m_bi.ulFlags = BIF_USENEWUI;
-	m_bi.lpfn = NULL;
-	m_bi.lParam = NULL;
-	m_bi.iImage = 0;
+	if (!SUCCEEDED(CoCreateInstance(CLSID_FileOpenDialog, NULL, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(&m_pFileDialog))))
+	{
+		return false;
+	}
+
+	// Set the options to select folders
+	m_pFileDialog->GetOptions(&m_openDialogOptions);
+	m_pFileDialog->SetOptions(m_openDialogOptions | FOS_PICKFOLDERS);
+	m_pFileDialog->SetTitle(m_title);
 
 	return true;
 }
 
 void PathDialogWin32::destroy()
 {
+	if (m_pFileDialog)
+	{
+		m_pFileDialog->Release();
+	}
```

Comment:
Set m_pFileDialog to nullptr to prevent issue if destroy is called multiple times.